OrbitCore/CaptureData.h

```diff
@@ -125,6 +129,10 @@ class CaptureData {
 
   [[nodiscard]] const std::shared_ptr<Process>& process() const { return process_; }
 
+  [[nodiscard]] const std::shared_ptr<SamplingProfiler>& sampling_profiler() const {
```

Comment:
should this be a const object? std::shared_ptr<const SamplingProfiler>?

Reply:
or have 2 functions - one that returns const pointer/const reference another (non-const) which returns shared_ptr.

Reply:
I added a second getter returning a const ref SamplingProfiler

Reply:
I think that if you're using this function to update profiler it shouldn't be `const`. Also I would prefer to use 
`[[nodiscard]] SamplingProfiler* sampling_profiler()` or 
`[[nodiscard]] std::shared_ptr<SamplingProfiler> sampling_profiler()` if you need shared_ptr.

Reply:
I'll make it a non-const ref, as it is also stored into the sampling report (and I want avoid storing raw pointer here).
In the future, this will be a value type and the second getter will go away.

Reply:
actually made it returning the shared_ptr as value.